Remotion/Data/DomainObjects.UnitTests/Persistence/PersistenceServiceTest.cs

```diff
@@ -568,7 +568,7 @@ public void Save_WithRollback_CallsExtensionPoints ()
       var isRollbackCalled = false;
       var contextMock = new Mock<IDisposable>(MockBehavior.Strict);
       contextMock.Setup(mock => mock.Dispose()).Callback(() => Assert.That(isRollbackCalled, Is.True)).Verifiable();
-      var storageProviderMock = new Mock<StorageProvider>(MockBehavior.Strict, UnitTestStorageProviderDefinition, NullPersistenceExtension.Instance);
+      var storageProviderMock = new Mock<IStorageProvider>(MockBehavior.Strict, UnitTestStorageProviderDefinition, NullPersistenceExtension.Instance);
```

Comment:
should we use IReadOnlyStorageProvider mocks where possible in this file?

Reply:
This is not straightforward, because `_innerMockStorageProvider` is a `SafeContextSingleton<IStorageProvider>` where we cannot set an `IReadOnlyStorageProvider`, and we cannot change it because sometimes we need the write-capability.
We could split it, but I don't really see the point.

Reply:
aaaah.... I see.